File: python/novax_sdk/pay/h5/h5_order_request.py
from dataclasses import dataclass
from typing import Any

from novax_sdk.pay.model import PayOrderResp
from novax_sdk.request.api_request import ApiRequest, HttpMethod


@dataclass
class H5OrderRequest(ApiRequest):
    """GET /pay/public/h5/order"""

    token: str | None = None

    @property
    def method(self) -> HttpMethod:
        return HttpMethod.GET

    @property
    def path(self) -> str:
        return "/pay/public/h5/order"

    def query_params(self) -> dict[str, Any]:
        return {k: v for k, v in {"token": self.token}.items() if v is not None}

    @property
    def response_type(self) -> type:
        return PayOrderResp

File: java/sdk/sdk-pay/src/main/java/com/novax/sdk/pay/h5/H5OrderRequest.java
package com.novax.sdk.pay.h5;

import com.novax.sdk.core.request.AbstractApiRequest;
import com.novax.sdk.core.request.HttpMethod;
import com.novax.sdk.core.request.TypeRef;
import com.novax.sdk.pay.model.PayOrderResp;
import lombok.Builder;
import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * GET /pay/public/h5/order — fetch the H5 payment order summary by token.
 */
@Getter
@Builder
public class H5OrderRequest extends AbstractApiRequest<PayOrderResp> {

    private final String token;

    @Override
    public HttpMethod method() {
        return HttpMethod.GET;
    }

    @Override
    public String path() {
        return "/pay/public/h5/order";
    }

    @Override
    public Map<String, ?> queryParams() {
        Map<String, Object> q = new LinkedHashMap<>();
        if (token != null) q.put("token", token);
        return q;
    }

    @Override
    public TypeRef<PayOrderResp> responseType() {
        return TypeRef.of(PayOrderResp.class);
    }
}
